src/main/java/org/openrewrite/java/logging/slf4j/Slf4jLogShouldBeConstant.java

```diff
@@ -120,21 +128,21 @@ public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext execu
         };
     }
 
+    private boolean containsCombinedFormatSpecifiers(String str) {
+        return FORMAT_SPECIFIER_PATTERN.matcher(str).find() && SLF4J_FORMAT_SPECIFIER_PATTERN.matcher(str).find();
+    }
+
     private static String replaceFormatSpecifier(String str, String replacement) {
         if (str == null || str.isEmpty()) {
             return str;
         }
-        Pattern pattern = Pattern.compile("%[\\d\\.]*[dfscbBhHn%]");
-        Matcher matcher = pattern.matcher(str);
-        return matcher.replaceAll(replacement);
+        return FORMAT_SPECIFIER_PATTERN.matcher(str).replaceAll(replacement);
     }
 
     private static boolean containsIndexFormatSpecifier(String str) {
         if (str == null || str.isEmpty()) {
             return false;
         }
-        // A regular expression that matches index specifiers like '%2$s', '%1$s', etc.
-        String indexSpecifierRegex = ".*%(\\d+\\$)[a-zA-Z].*";
-        return str.matches(indexSpecifierRegex);
+        return INDEXED_FORMAT_SPECIFIER_PATTERN.matcher(str).find();
```

Comment:
There's a subtle change here that I'm not sure was intentional, so I'm pointing it out. We went from `String#matches(String)`, which is equivalent to `Pattern.compile(regex).matcher(input).matches()`, to `Pattern.compile(regex).matcher(input).find()`.

`Matcher#matches()` attempts to match the entire region against the pattern, whereas
`Matcher#find()` attempts to find the next subsequence of the input sequence that matches the pattern.

Now in this case the regex starts and ends with `.*`, which should make these fairly equivalent if I'm not mistaken, but perhaps it's good to be clear about our intention here. Do we want to use `find()`, and potentially lose the leading/trailing wiildcards? Or we want to keep `matches()` with the wildcards?

Reply:
good find! (pun, only slighly, intended)
With the wildcards in the regex, the result will be the same, just a bit slow I think. Stripping off the wildcards from the regex will be the more readable solution I think, performance wise it should not matter (much). Since the method name uses "contains" and matches implies equality not contains.

Do you agree?

Reply:
Agreed!